### ISSUE TITLE
Too many challenges are doable with simple implementation

### ISSUE THREAD

**Issue by magwo** [magwo/elevatorsaga#4]
The challenges should scale more quickly to block any simple implementations from getting far.

Perhaps an early move efficiency challenge.

**Comment by ArcanisCz** [comment]
Simple idea, out of blue and not tested - what about adding underground garagte level(s)?
Second idea based on real life struggles - some people want to go up, but push "down" button in hall. (but they will press correct floor button inside elevator)

**Comment by magwo** [comment]
First idea is pretty good. It's essentially the same thing as a regular building but with assymetric spawning patterns - most people  appearing on, for example floor 3.

Second idea is a little interesting - it has been mentioned in other tickets about making passenger behaviour a little random and erratic. Perhaps it adds enjoyment to the game.

**Comment by magwo** [comment]
With all game ideas I want there to be a good ratio between complexity and interest factor.. if something makes gameplay and implementation much more complex without actually adding any interesting challenge, then it is not a good idea.

**Comment by ArcanisCz** [comment]
yea, i agree that irrational people behaviour could be interesting, but quite complex to implement it right and then handle it in game....

Maybe i had third idea - what about some "disaster" like fire in some floor(s)? Like "now, many people needs to be transported from floor 3". Which will penalize all implementation, which goes to floor 0 by default

**Comment by magwo** [comment]
Yep that's an interesting concept - changing spawning patterns during a challenge.

**Comment by roryokane** [comment]
To add more detail, here is one “simple implementation” that sometimes passes challenges 2 and 3, but shouldn’t be able to.

```javascript
{
    init: function(elevators, floors) {
        var elevator = elevators[0];
        elevator.on("idle", function() {
            elevator.goToFloor(0);
            elevator.goToFloor(1);
            elevator.goToFloor(2);
            elevator.goToFloor(3);
            elevator.goToFloor(4);
        });
    },
    update: function(dt, elevators, floors) {
        // We normally don't need to do anything here
    }
}
```

**Comment by codeholic** [comment]
> Maybe i had third idea - what about some "disaster" like fire in some floor(s)?

One is not allowed to use elevators in case of fire. Just saying :innocent:

**Comment by nivoset** [comment]
c ould also add elevator breaking or being pulled out of service
